### DISCUSSION TITLE
Fatal JS error: SecurityError: Attempt to use history.replaceState() more than 100 times per 30 seconds

### DISCUSSION THREAD

**Discussion by ralphjsmit** [livewire/livewire#5923] [7. Bugs (archived)]
I am running a Filament beta app on a staging environment. Filament V3 uses the `wire:navigation` feature.

If I quickly click through 7 pages (e.g. via the sidebar), then the following error will happen:

<img width="1062" alt="Scherm­afbeelding 2023-07-24 om 11 45 12" src="https://github.com/livewire/livewire/assets/59207045/858e96a9-0131-48ab-8220-f39faa504fea">

The page will be a gray page and not work:

<img width="1262" alt="Scherm­afbeelding 2023-07-24 om 11 45 28" src="https://github.com/livewire/livewire/assets/59207045/e1173e06-80ce-450f-b8e6-80a9acfb6bfc">

**Comment by joshhanley** [comment]
@ralphjsmit thanks for reporting! Ah that's an interesting one, I did not know there was a limit. I will get Caleb to have a look when he gets the chance.

**Comment by joshhanley** [comment]
@ralphjsmit I'm currently trying to look further into this, but I can't seem to replicate it. I did a test with 300 clicks in 30 seconds and it didn't error out.

<img width="737" alt="image" src="https://github.com/livewire/livewire/assets/882837/e5d2f57d-ac52-457a-8ea5-4bdfaed9ae41">

Any chance you could create a small example repo that demonstrates the issue - preferably without Filament, but if you can't get it to work, then use it - so we can investigate it further? Thanks!

**Comment by ralphjsmit** [comment]
@joshhanley ~~Thanks for looking into this! I just checked again in Safari and I cannot reproduce this anymore. Seems like this then is already fixed, so closing the issue. Thanks!~~

**Update:** Sorry, I forgot that Filament (temporarily) dropped the `wire:navigate` feature, so that obviously also 'fixed' the issue. I'll try to create an example repo.

**Comment by HDVinnie** [comment]
Im also facing this only in safari after updating from Livewire 2 to 3. 

https://github.com/HDInnovations/UNIT3D-Community-Edition/pull/3608

Not using wire:navigate. 

<img width="1067" alt="Screen Shot 2024-03-04 at 2 06 41 PM" src="https://github.com/livewire/livewire/assets/12850699/6d0a9d5e-9850-4067-9c26-2c2012e9b854">

**Comment by Capevace** [comment]
I've ran into the same issue, it appears to be calling replaceState about 10-20 times when clicking a `wire:navigate` link. I can't get it to reproduce in an empty repo tho.

I've managed a hotfix which replaces replaceState with a throttled version. Haven't played around with the timings too much but I haven't had any problems after adding this.

```js
const original = window.history.replaceState;
let timer = Date.now();

let timeout = null;
let lastArgs = null;

window.history.replaceState = function (...args) {
	const time = Date.now();

	if (time - timer < 300) {
		lastArgs = args;
		
		if (timeout) {
			clearTimeout(timeout);
		}
		
		timeout = setTimeout(() => {
			original.apply(this, lastArgs);
			
			timeout = null;
			lastArgs = null;
		}, 100);
		
		return;
	}

	timer = time;

	original.apply(this, args);
};
```

If you want to use this, keep in mind this will drop throttled replaceState calls and tries to only apply the latest state, so you *may* lose some state, idk `¯\_(ツ)_/¯`

**Reply by nezaboravi** [reply]
Got the same issue

Where did you place this piece of code @Capevace ?

**Reply by Capevace** [reply]
> Got the same issue
> 
> 
> 
> Where did you place this piece of code @Capevace ?
> 
> 

Somewhere global, I put it in my main JS file I load everywhere.

**Comment by lcamero** [comment]
What I've been able to find out is that it is related to the `#[Url]` attribute used by Livewire. I have a use case in my FilamentPHP project where I have a ParentComponent rendering multiple ChildComponents that use the `InteractsWithActions` trait from FilamentPHP which creates a couple of properties that use the attribute.

I wrote down more details in this other thread #7746 ([my comment](https://github.com/livewire/livewire/discussions/7746#discussioncomment-10389123)) but it essentially seems to boil down to that. I have not been able to figure out what Filament does in something like the tables where they render each row with actions on them (which is similar to what I'm doing) and not get the same error

@Capevace 's workaround seems to alleviate the issue but it can certainly have unwanted side effects in some cases. I believe the attribute just runs a replace X times as components you have, which the throttle solution probably handles relatively well

**Comment by juliancc** [comment]
Has anyone encounter a way to properly fixed this? @Capevace does your workaround created any unwanted side effects as @lcamero suggested... I just ran into this problem and wanted to see what the best approach

**Reply by Capevace** [reply]
Im not 100% sure we're still using it, but I haven't had any issues with it personally

**Reply by CalumBThomson** [reply]
@juliancc - I believe [this PR](https://github.com/livewire/livewire/pull/9059) by @luanfreitasdev addresses this issue cleanly by only performing the replaceState action if location / history has actually changed.

**Reply by juliancc** [reply]
I see it now. Subscribed to it and following.. thanks

**Comment by patie** [comment]
same problem here :(

**Comment by robinvda** [comment]
Got the same issue. As mentioned by @lcamero this seems to be caused by the [`#[Url]` attribute](https://livewire.laravel.com/docs/url).

I've reproduced this in a clean Laravel/Livewire install [here](https://github.com/robinvda/livewire-navigation-throttle-issue) (quick and dirty ;), resulting in this:

Chrome (after navigating to settings/welcome a few times)
<img width="3124" height="1592" alt="CleanShot 2025-08-13 at 09 24 38@2x" src="https://github.com/user-attachments/assets/e4da9d6e-f3cb-4997-accd-d00ada18b62a" />

Safari (after navigating twice)
<img width="1870" height="640" alt="CleanShot 2025-08-13 at 09 19 58@2x" src="https://github.com/user-attachments/assets/8e1ae9ee-cafb-4579-a0f1-176030b30363" />

The main issue for me is that the url stops changing after the console error is thrown (at least in Chrome), so refreshing the page then brings you back to a different page than the one you're currently on.
